DS and Algorithm/Day1/LinkedList.py

```diff
@@ -0,0 +1,53 @@
+class Node:
+    def __init__(self, data):
+        self.data = data
+        self.next = None
+
+def reverse_linked_list(head):
+    # Initialize pointers to keep track of the current, previous, and next nodes
+    prev = None            # To keep track of the previous node
+    current = head         # To traverse the linked list starting from the head
+
+    # Traverse the linked list
+    while current is not None:
+        # Store the next node before changing the 'next' reference
+        next_node = current.next
+
+        # Reverse the link by pointing the 'next' of the current node to the previous node
+        current.next = prev
+
+        # Move to the next pair of nodes in the list
+        prev = current
+        current = next_node
+
+    # The 'prev' pointer now points to the new head of the reversed linked list
+    head = prev
+
+    # Return the new head of the reversed linked list
+    return head
```

Comment:
**Optimization Issue**: The 'reverse_linked_list' function uses an iterative approach which is efficient, but there's an opportunity to reduce the number of lines and improve readability by simplifying the swapping logic. <br> **Fix**: Simplify the swapping logic within the 'reverse_linked_list' function to improve code readability without compromising the efficiency of the algorithm. <br> **Code Suggestion**: 
 ```
 def reverse_linked_list(head):
     prev, current = None, head
     while current:
         current.next, prev, current = prev, current, current.next
     return prev
 ```